repo/repo.go

```diff
@@ -648,6 +648,16 @@ func tempFile(fn string, perm os.FileMode) (f *os.File, err error) {
 	return
 }
 
+func syncFile(f *os.File) error {
+	err := f.Sync()
+	// Ignore error if filesystem does not support fsync.
+	// macOS Ventura returns "inappropriate ioctl for device" in some cases
+	if errors.Is(err, syscall.ENOTSUP) || errors.Is(err, syscall.ENOTTY) {
```

Comment:
That part is copied from https://github.com/restic/restic/blob/8dd95b710e6ba2a267c27f7ad0087d68cad7bdfd/internal/backend/local/local.go#L180 and https://github.com/restic/restic/blob/8dd95b710e6ba2a267c27f7ad0087d68cad7bdfd/internal/backend/local/local_unix.go#L22 . Apparently the errnos when fsyncing directories are much more chaotic than for files.

Reply:
Would it make sense to log unexpected (not in this list) errors instead of returning them to the callers and failing the whole operation?

Another option would be to log an error for any error code, but only once. This could at least warn the user that the data may not be flushed on this type of device.

fsync is indeed a mess full of false promises anyway: https://lwn.net/Articles/752063/

Reply:
Ignoring the errnos unless we know that they are legitimate sounds like a bad idea. rest-server should not claim that it stored data unless we're pretty sure that it did.

> fsync is indeed a mess full of false promises anyway: https://lwn.net/Articles/752063/

I don't think the scenario from the LWN article applies for us. rest-server creates a new file, writes data and calls fsync on it. If any of these operations fail, then the file is discarded.

> Another option would be to log an error for any error code, but only once. This could at least warn the user that the data may not be flushed on this type of device.

Logging a warning on the rest-server side when fsync is not supported would be an option.

Reply:
I've added a warning message to is printed once.